app/models/apple/publisher.rb

```diff
@@ -107,41 +107,49 @@ def upload_and_process!(eps)
     end
 
     def upload_media!(eps)
-      # Soft delete any existing delivery and delivery files.
-      prepare_for_delivery!(eps)
+      Rails.logger.tagged("Apple::Publisher##{__method__}") do
+        Rails.logger.info("Starting media upload", {episode_count: eps.length})
+
+        # Soft delete any existing delivery and delivery files.
+        prepare_for_delivery!(eps)
 
-      # Only create if needed.
-      sync_episodes!(eps)
-      sync_podcast_containers!(eps)
+        # Only create if needed.
+        sync_episodes!(eps)
+        sync_podcast_containers!(eps)
 
-      wait_for_versioned_source_metadata(eps)
+        wait_for_versioned_source_metadata(eps)
 
-      sync_podcast_deliveries!(eps)
-      sync_podcast_delivery_files!(eps)
+        sync_podcast_deliveries!(eps)
+        sync_podcast_delivery_files!(eps)
 
-      # Upload and mark as uploaded, then update the audio container reference.
-      execute_upload_operations!(eps)
-      mark_delivery_files_uploaded!(eps)
-      update_audio_container_reference!(eps)
+        # Upload and mark as uploaded, then update the audio container reference.
+        execute_upload_operations!(eps)
+        mark_delivery_files_uploaded!(eps)
+        update_audio_container_reference!(eps)
 
-      # Mark the episode as uploaded.
-      mark_as_uploaded!(eps)
+        # Mark the episode as uploaded.
+        mark_as_uploaded!(eps)
 
-      # The episodes start waiting after they are uploaded.
-      # Increment the wait counter.
-      increment_asset_wait!(eps)
+        # The episodes start waiting after they are uploaded.
+        # Increment the wait counter.
+        increment_asset_wait!(eps)
+      end
     end
 
     def process_delivery!(eps)
-      increment_asset_wait!(eps)
+      Rails.logger.tagged("Apple::Publisher##{__method__}") do
+        Rails.logger.info("Starting delivery processing", {episode_count: eps.length})
 
-      wait_for_upload_processing(eps)
+        increment_asset_wait!(eps)
 
-      # Wait for the audio asset to be processed by Apple
-      # Mark episodes as delivered as they are processed
-      wait_for_asset_state(eps) do |ready_eps|
-        mark_as_delivered!(ready_eps)
-        reset_asset_wait!(ready_eps)
+        wait_for_upload_processing(eps)
+
+        # Wait for the audio asset to be processed by Apple
+        # Mark episodes as delivered as they are processed
+        wait_for_asset_state(eps) do |ready_eps|
+          log_asset_wait_duration!(ready_eps)
```

Comment:
So besides the tagged logger block, this is the difference here - `log_asset_wait_duration!` instead of `reset_asset_wait!`, and do it before marking delivered